src/go/rdctl/pkg/factoryreset/delete_data.go

```diff
@@ -0,0 +1,340 @@
+		"windows": unregisterAndDeleteWindowsData,
+	}[runtime.GOOS](removeKubernetesCache)
+}
+
+func getStandardDirs() (string, string, string, error) {
+	configDir, err := os.UserConfigDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return "", "", "", err
+	}
+	return configDir, cacheDir, homeDir, nil
+}
+
+func deleteDarwinData(removeKubernetesCache bool) error {
+	configDir, cacheDir, homeDir, err := getStandardDirs()
+	if err != nil {
+		return err
+	}
+	libraryPath := path.Join(homeDir, "Library")
+
+	altAppHomePath := path.Join(homeDir, ".rd")
+	appHomePath := path.Join(configDir, "rancher-desktop")
+	cachePath := path.Join(cacheDir, "rancher-desktop")
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath == "" {
+		logsPath = path.Join(libraryPath, "Logs", "rancher-desktop")
+	}
+	settingsPath := path.Join(libraryPath, "Preferences", "rancher-desktop")
+	updaterPath := path.Join(configDir, "Caches", "rancher-desktop-updater")
+
+	pathList := []string{
+		altAppHomePath,
+		appHomePath,
+		logsPath,
+		settingsPath,
+		updaterPath,
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteLinuxLikeData(homeDir, altAppHomePath, path.Join(homeDir, ".config"), pathList)
+}
+
+func deleteLinuxData(removeKubernetesCache bool) error {
+	configHomePath, cacheHomePath, homeDir, err := getStandardDirs()
+	if err != nil {
+		return err
+	}
+
+	dataDir := os.Getenv("XDG_DATA_HOME")
+	if dataDir == "" {
+		dataDir = path.Join(homeDir, ".local", "share")
+	}
+	altAppHomePath := path.Join(homeDir, ".rd")
+	cachePath := path.Join(cacheHomePath, "rancher-desktop")
+	configPath := path.Join(configHomePath, "rancher-desktop")
+	dataHomePath := path.Join(dataDir, "rancher-desktop")
+
+	pathList := []string{
+		altAppHomePath,
+		configPath,
+		path.Join(homeDir, ".local", "state", "rancher-desktop"),
+	}
+	logsPath := os.Getenv("RD_LOGS_DIR")
+	if logsPath != "" {
+		pathList = append(pathList, logsPath, path.Join(dataHomePath, "lima"))
+	} else {
+		pathList = append(pathList, path.Join(dataHomePath))
+	}
+	if removeKubernetesCache {
+		pathList = append(pathList, cachePath)
+	}
+	return deleteLinuxLikeData(homeDir, altAppHomePath, configHomePath, pathList)
+}
+
+func unregisterAndDeleteWindowsData(removeKubernetesCache bool) error {
+	err := unregisterWSL()
+	if err != nil {
+		return err
+	}
+	err = deleteWindowsData(!removeKubernetesCache, "rancher-desktop")
+	if err != nil {
+		return err
+	}
+	return clearDockerContext()
+}
+
+func deleteLinuxLikeData(homeDir string, altAppHomePath string, configHomePath string, pathList []string) error {
+	err := deleteLimaVM()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error trying to delete the Lima VM: %s\n", err)
+	}
+	for _, currentPath := range pathList {
+		err := os.RemoveAll(currentPath)
+		if err != nil {
```

Comment:
I'm not a fan of this shortcut because it only works for functions that return only `error`.  If you've defined
a function that only returns an `error`, and then for some reason it needs to return something like `(foo, error)`
you have to change more than one line of code.

But I wrote a ruby 3-liner (couldn't figure out how to do it on the command-line) and applied it to all the files in go/rdctl:
```
contents = ARGF.read

ptn = /^(\s+)(err :?=.*)\n\1if err != nil {/

puts contents.gsub(ptn) { |m| "#{$1}if #{$2}; err != nil {"}
```

You still need to manually check for subsequent `err`s that become undefined because of the change in scope.

I'll try to remember not to squash the commit with this change, so it's easily undoable if we need to do that.